src/vs/platform/terminal/node/ptyService.ts

```diff
@@ -29,12 +33,21 @@ export class PtyService extends Disposable implements IPtyService {
 	readonly onProcessOverrideDimensions = this._onProcessOverrideDimensions.event;
 	private readonly _onProcessResolvedShellLaunchConfig = this._register(new Emitter<{ id: number, event: IShellLaunchConfig }>());
 	readonly onProcessResolvedShellLaunchConfig = this._onProcessResolvedShellLaunchConfig.event;
-
 	constructor(
 		private readonly _logService: ILogService
 	) {
 		super();
 	}
+	onPtyHostExit?: Event<number> | undefined;
+	onPtyHostStart?: Event<void> | undefined;
+
+	acknowledgeDataEvent(id: number, charCount: number): Promise<void> {
+		throw new Error('Method not implemented.');
```

Comment:
This is probably just making it async and `this._throwIfNoPty(id).acknowledgeDataEvent(charCount);`